src/internal/packager/helm/post-render.go

```diff
@@ -33,16 +35,24 @@ type renderer struct {
 func (h *Helm) newRenderer() (*renderer, error) {
 	message.Debugf("helm.NewRenderer()")
 
-	namespaces := make(map[string]*corev1.Namespace)
-	if h.cluster != nil {
-		namespaces[h.chart.Namespace] = h.cluster.NewZarfManagedNamespace(h.chart.Namespace)
-	}
-
-	return &renderer{
+	rend := &renderer{
 		Helm:           h,
-		connectStrings: make(types.ConnectStrings),
-		namespaces:     namespaces,
-	}, nil
+		connectStrings: types.ConnectStrings{},
+		namespaces:     map[string]*corev1.Namespace{},
+	}
+	if h.cluster == nil {
+		return rend, nil
+	}
+	namespace, err := h.cluster.K8s.GetNamespace(h.chart.Namespace)
```

Comment:
We don't have to explicitly call `K8s` here because it is composed into `Cluster`:

```suggestion
	namespace, err := h.cluster.GetNamespace(h.chart.Namespace)
```